### DISCUSSION TITLE
希望能增强公共构建机 【登录调试】页面功能。支持文件浏览、上传、下载等功能。

### DISCUSSION THREAD

**Discussion by mario206** [Tencent/bk-ci#6466] [General]
原因：
1. 文件浏览：很多开发者平时主要使用windows/mac，不熟悉 shell 命令，上机器浏览文件、跳转目录等很麻烦。比如列出文件大小需要 ls -lh，就得 google 一圈才知道，效率很低。
2. 上传：有时需要编辑文件，在本地 IDE 环境下编辑，然后上传，会更方便
3. 下载：有时需要查看文件，很多文件是二进制格式，控制台无法查看。即使是文本文件，对于大的文件浏览也不方便。

**Comment by zanyzhao** [comment]
登录调试功能无法做到“CI定义可追踪”，如果是要通过登录调试改很多才能跑通的CI流程，建议直接修改流水线。

**Reply by mario206** [reply]
什么是 【CI定义可追踪】？

**Reply by zanyzhao** [reply]
CI定义可追踪：我的理解是，CI定义的过程必须可追溯，登录调试的功能是手动进行的，假如编译失败，然后通过登录调试编译成功，从失败到成功，平台无法捕获你到底做了什么，时间久了，估计你自己也会忘了。
所以，不要太依赖登录调试，而是通过流水线将CI过程描述清楚，这样便于维护和传播。

**Comment by mario206** [comment]
文件浏览： 列出文件名、大小、修改时间等一些常用信息
上传：主要是为了调试一些新写的脚本。编辑->上传->运行，循环直到成功运行后，再提交仓库，这会加速开发效率。因为目前的流水线很复杂（各种插件/网络环境），本地环境已经无法模拟，所以一些脚本直接在流水线上编写、调试。希望设置好两个路径后，一键同步
下载：主要是方便在本机查看日志、缓存等，有时是整个文件夹(可能需要zip)。最好选中后直接页面下载
